### PR TITLE
[CST-5449] "Browse by" pages are missing "Now showing" contextual information

### LINKED ISSUE #369: [Usability issue] "Browse by" pages are missing "Now showing" contextual information
After merging #364, all Browse By pages are now missing important contextual information.  

Namely, the "Now showing items __ of ___" text is missing on all pages.  This text can be seen in the XMLUI at: http://demo.dspace.org/xmlui/browse?type=title 

This has several major usability problems:
* The user cannot has no sense of how many items are in the "browse by" results (i.e. no sense of how many pages exist)
* Pagination becomes incredibly confusing, as you have no idea what page you are on, how many pages exist, and whether pagination is even _functioning properly_ (as the only sign you are on the last page is that the next page button suddenly grays out).

This contextual information was removed by the addition of the "Starts-With" component in #364

### PR DESCRIPTION
**Pull request by davide-negretti**
## References
* Fixes #369 

## Description

"Now showing" has been added to "browse by" pages also when showPaginator is false

## Checklist

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Comment by MarieVerdonck** [comment]
You have to merge in the latest community main to even render this branch (otherwise error:

```
zone-evergreen.js:659 Unhandled Promise rejection: Cannot read properties of undefined (reading 'nameSpace') ; Zone: <root> ; Task: Promise.then ; Value: TypeError: Cannot read properties of undefined (reading 'nameSpace')
    at Object.getBase [as useFactory] (app.module.ts:68:23)
```

The results of a page are present twice, with the second being in a narrower div, centered on page, causing an indent. Also the previous/next buttons then are left/right aligned within this indented div of second half of results (see screenshot)

Considering the complaints from the [[original issue](https://github.com/DSpace/dspace-angular/issues/369)](https://github.com/DSpace/dspace-angular/issues/369) boil down to:

- No "Now showing items __ of ___"
- Page of pagination not clear like on regular pagination buttons (<<|1|2|3|...|{lastPagNr}|>>)

I would undo these changes and just set enableArrows to false [[here](https://github.com/DSpace/dspace-angular/pull/1552/files#diff-7fdd79d04dea64a9b478b659fc6097552fe2b2d99790708a765c589fce3babfeL34)](https://github.com/DSpace/dspace-angular/pull/1552/files#diff-7fdd79d04dea64a9b478b659fc6097552fe2b2d99790708a765c589fce3babfeL34). 

Browse already has two optional views built in: the one currently on demo (enabledArrowse=true), and the regular pagination one (enableArrows=false), so with "Now showing items __ of ___" and pagination buttons. The latter of which uses shared `ds-viewable-collection`, the former is custom to the browse pages.

The 'enableArrows=true' view was likely added since this is more in line with the old view on DS6 ([[example](https://demo.dspace.org/xmlui/browse?resetOffset=true)](https://demo.dspace.org/xmlui/browse?resetOffset=true)), but the pagination arrows include more info on pagination and allow you to skip multiple pages at once, so seems much more user friendly which would be an improvement.

In regards with compatibility with the suggested changes in https://github.com/DSpace/dspace-angular/pull/1561 if we go for just the enableArrows=false change, then the change there like showing tooltip on disabled next button would still make the 'enableArrows=true' view slighly more user friendly, but since this view wouldn't be default, it can be removed there if requested.

File: src/app/browse-by/browse-by-metadata-page/browse-by-metadata-page.component.html [hunk 1]
```diff
@@ -31,7 +31,6 @@
       [sortConfig]="(currentSort$ |async)"
       [type]="startsWithType"
       [startsWithOptions]="startsWithOptions"
-      [enableArrows]="true"
```
Comment (MarieVerdonck) [inline src/app/browse-by/browse-by-metadata-page/browse-by-metadata-page.component.html:34]: So I would just set enableArrows=false here, and undo other changes

File: src/app/shared/browse-by/browse-by.component.html [hunk 2]
```diff
@@ -1,42 +1,20 @@
...
-        <li *ngFor="let object of objects?.payload?.page" class="mt-4 mb-4">
-          <ds-listable-object-component-loader [object]="object" [viewMode]="viewMode"></ds-listable-object-component-loader>
-        </li>
-      </ul>
-      <div>
-        <button id="nav-prev" type="button" class="btn btn-outline-primary float-left" (click)="goPrev()" [disabled]="objects?.payload?.currentPage <= 1"><i class="fas fa-angle-left"></i> {{'browse.previous.button' |translate}}</button>
-        <button id="nav-next" type="button" class="btn btn-outline-primary float-right" (click)="goNext()" [disabled]="objects?.payload?.currentPage >= objects?.payload?.totalPages"><i class="fas fa-angle-right"></i> {{'browse.next.button' |translate}}</button>
-      </div>
-    </div>
+    <ds-viewable-collection
```
Comment (MarieVerdonck) [inline src/app/shared/browse-by/browse-by.component.html:5]: You could keep these changes in ds-viewable-collection (provided the duplicate page with indent rendering is fixed) and enable the alternate arrow view with enableArrows boolean there.
But the 1|2|3 pagination buttons are more user friendly, so by default I would use those.

**Review by MarieVerdonck — CHANGES_REQUESTED** [review]

**Comment by atarix83** [comment]
@tdonohue @MarieVerdonck  could you check again, the issue should be resolved now.

I left the enableArrows to true so you can test the behaviour with next and prev button, in the #1561 can be turned to false

**Review by tdonohue — APPROVED** [review]
👍 Thanks @davidenegretti-4science and @atarix83 !  Tested this today and it works as described.  All Browse by pages have "Now Showing..." at the top. 

Per @MarieVerdonck 's feedback about using the 1|2|3 pagination instead of the Previous/Next buttons: As discussed with @artlowel in yesterday's meeting, this PR can go forward with the Previous/Next button approach.  As we mentioned though, @MarieVerdonck can always switch it over to using 1|2|3 pagination in PR #1561 -- as I do agree that, if it's easy to achieve, the 1|2|3 pagination may provide us with a better user experience.

All in all, I think this PR looks good enough to move forward.  @MarieVerdonck: Did you have any final comments to add?  I believe all your feedback/bugs have been resolved (other than the 1|2|3 pagination, obviously).  But, I'll give you a chance to give this one last look.

File: src/app/shared/browse-by/browse-by.component.html [hunk 3]
```diff
@@ -1,42 +1,20 @@
...
+      [config]="paginationConfig"
+      [sortConfig]="sortConfig"
+      [showPaginator]="showPaginator"
```
Comment (MarieVerdonck) [inline src/app/shared/browse-by/browse-by.component.html:8]: Setting the showPaginator to true here seems to enable the 123 view in browse

**Review by MarieVerdonck — APPROVED** [review]
Previous issue seems fixed and verified you can now go to the 123 view again, now with the `[showPaginator]="true"` of `ds-viewable-collection` in `browse-by.component.html`. Feel free to merge so we can rebase in [#1561](https://github.com/DSpace/dspace-angular/pull/1561) so we can see how these changes interact with those (location go back button / tooltip on last next button might not work any more/need changing). Afterwards we can enable the 123 view by default if everyone agrees this provides the better UX concerning pagination.

**Comment by tdonohue** [comment]
Merging as this is at +2.  Thanks @davide-negretti, and @pratik-peerbits and @atarix83 !

**Pull request merged**